floc.bs

```diff
@@ -1 +1,169 @@
+Editor: Josh Karlin, Google, jkarlin@chromium.org
+Abstract: This specification describes a method that could enable ad-targeting based on the people’s general browsing interest without exposing the exact browsing history.
+!Participate: <a href="https://github.com/WICG/floc">GitHub WICG/floc</a> (<a href="https://github.com/WICG/floc/issues/new">new issue</a>, <a href="https://github.com/WICG/floc/issues?state=open">open issues</a>)
+Group: wicg
+Repository: WICG/floc
+</pre>
 
+<pre class=link-defaults>
+spec:html; type:attribute; text:document
+</pre>
+
+<section>
+  <h2 id='introduction'>Introduction</h2>
+
+  In today's web, people’s interests are typically inferred based on observing what sites or pages they visit, which relies on tracking techniques like third-party cookies or less-transparent mechanisms like device fingerprinting. It would be better for privacy if interest-based advertising could be accomplished without needing to collect a particular individual’s browsing history.
+
+  This specification provides an API to enable ad-targeting based on the people’s general browsing interest, without exposing the exact browsing history.
+
+  <div class="example">
+    Creating an ad based on the interest cohort:
+
+    <pre class="lang-js">
+      const cohort = await document.interestCohort();
+      const url = new URL("https://ads.example/getCreative");
+      url.searchParams.append("cohort_id", cohort.id);
+      url.searchParams.append("cohort_version", cohort.version);
+      const creative = await fetch(url);
+    </pre>
+  </div>
+</section>
+
+<section>
+  <h2 id="interest-cohort-section">Interest cohort</h2>
+
+  The <dfn>interest cohort</dfn> is a user's assigned interest group under a particular <a href="#cohort-assignment-algorithm">cohort assignment algorithm</a>. An interest cohort comprises an [=interest cohort id=] and an [=interest cohort version=].
+
+  The <dfn>interest cohort id</dfn> represents the interest group that the user is assigned to by the <a href="#cohort-assignment-algorithm">cohort assignment algorithm</a>. The total number of groups should not exceed 2^32, and each group can mapped to a 32 bit integer. The interest cohort id can be invalid, which means no group is assigned.
+
+  The <dfn>string representation of the interest cohort id</dfn> is the string representation of the mapped integer of the [=interest cohort id=] in decimal (e.g. “17319”). If the [=interest cohort id=] is invalid, the string representation will be an empty string.
+
+  The <dfn>interest cohort version</dfn> identifies the <a href="#cohort-assignment-algorithm">algorithm</a> used to compute the [=interest cohort id=].
+
+  The <dfn>string representation of the interest cohort version</dfn> is browser defined. It’s recommended to make the browser vendor name as part of the version (e.g. “chrome.2.1”, “v21/mozilla”), so that when exposed to the Web, there won't be naming collisions across browser vendors.
```

Comment:
```suggestion
  The <dfn>string representation of the interest cohort version</dfn> is [=implementation-defined=]. It’s recommended to make the browser vendor name as part of the version (e.g. “chrome.2.1”, “v21/mozilla”), so that when exposed to the Web, there won't be naming collisions across browser vendors.
```

Reply:
Done.